src/late_multicellular_stage/Scalis.cs

```diff
@@ -0,0 +1,309 @@
+
+                    float minRadius = Mathf.Min(point.Radius, point.Parent.Radius);
+
+                    if (minRadius > 0.5f && boneDistances[j] < InnerCutoffPointMultiplier * minRadius / SurfaceValue)
+                    {
+                        return 10.0f;
+                    }
+                }
+            }
+        }
+
+        for (int j = 0; j < points.Length; j++)
+        {
+            var point = points[j];
+
+            if (point.Parent == null && points.Length > 1)
+                continue;
+
+            float aRadius = point.Radius;
+            float bRadius;
+
+            if (points.Length == 1)
+            {
+                bRadius = point.Radius;
+            }
+            else
+            {
+                bRadius = point.Parent!.Radius;
+            }
+
+            if (Cutoff && boneDistances[j] - additiveValue > Mathf.Pow(Mathf.Max(aRadius, bRadius)
+                    * CutoffPointMultiplier * 2.0f / SurfaceValue, 2.0f))
+            {
+                continue;
+            }
+
+            Vector3 a = point.Position / sigma;
+            Vector3 b;
+
+            if (points.Length == 1)
+            {
+                b = point.Position / sigma;
+                a.X -= aRadius;
+                b.X += aRadius;
+            }
+            else
+            {
+                b = point.Parent!.Position / sigma;
+            }
+
+            float tau0 = aRadius > bRadius ? bRadius : aRadius;
+            float deltaTau = Mathf.Abs(aRadius - bRadius);
+
+            if (aRadius > bRadius)
+            {
+                (a, b) = (b, a);
+            }
+
+            // Since i is always 3, we can cut down on some calculations and make static coeff array
+            // var coeff = BinomialExpansion(i - 1);
+
+            for (int k = 0; k < coefficients.Length; k++)
+            {
+                value += coefficients[k] * Mathf.Pow(deltaTau, k) * Mathf.Pow(tau0, i - k - 1)
+                    * Convolution(k, i, a, b, pos);
+            }
+        }
+
+        return value / NormalizationFactor(i, sigma);
+    }
+
+    public Color GetColour(Vector3 pos)
+    {
+        if (points.Length == 1)
+            return points[0].Colour;
+
+        Color colourSum = Colors.Black;
+
+        float contributionSum = 0.0f;
+
+        foreach (var point in points)
+        {
+            if (point.Parent == null)
+                continue;
+
+            float abDistanceSquared = point.Position.DistanceSquaredTo(point.Parent.Position);
+
+            Vector3 linePos = ClosestPoint(pos, point.Position, point.Parent.Position);
+
+            float contribution = 1.0f / linePos.DistanceSquaredTo(pos);
+
+            contributionSum += contribution;
+
+            colourSum += point.Colour.Lerp(point.Parent.Colour, point.Position.DistanceSquaredTo(linePos)
+                / abDistanceSquared) * contribution;
+        }
+
+        colourSum /= contributionSum;
+
+        colourSum.A = 0.0f;
```

Comment:
I assume that this is a typo as I assume surface colour shouldn't be fully transparent...